src/qibo/validation/gate_set_tomography.py

```diff
@@ -0,0 +1,2089 @@
+
+            elif class_qubit_gate is not None:
+                filedirectory = "GST_gates_qibo_1qb/"
+                if theta is None:
+                    fname = "%s gate (k=%.2d,j=%.2d) %s_basis.txt" % (
+                        class_qubit_gate,
+                        k,
+                        j,
+                        idx_basis[j],
+                    )
+                elif theta is not None:
+                    fname = "%s(%.12f) gate (k=%.2d,j=%.2d) %s_basis.txt" % (
+                        class_qubit_gate,
+                        theta,
+                        k,
+                        j,
+                        idx_basis[j],
+                    )
+                GST_gates_qibo_1qb = "GST_gates_qibo_1qb"
+                if not os.path.exists(GST_gates_qibo_1qb):
+                    os.makedirs(GST_gates_qibo_1qb)
+
+            # Save counts into a matrix. 0th column: word; 1st column: count value
+            # --------------------------------------------------------------------
+            fout = filedirectory + fname
+            fo = open(fout, "w")
+            for v1, v2 in counts.items():
+                fo.write(str(v1) + "  " + str(v2) + "\n")
+            fo.close()
+
+        # Find value for matrix_jk
+        # -------------------------
+
+        expectation_signs = np.matrix([[1, 1, 1, 1], [1, -1, -1, -1]])
+
+        matrix = np.zeros((2, 2))
+        for ii in range(0, 2):
+            word = bin(ii)[2:]
+            matrix[ii, 0] = word
+
+        for v1, v2 in counts.items():
+            # print(v1, v2)
+            row = int(v1, 2)
+            col = v2
+            matrix[row, 1] = col
+
+        sorted_matrix = sort_counts_1qb(matrix)
+        probs = sorted_matrix[:, 1] / np.sum(sorted_matrix[:, 1])
+        sorted_matrix = np.column_stack((sorted_matrix, probs))
+
+        k = k + 1
+        j = j + 1
+
+        if j == 1:
+            sorted_matrix = np.column_stack((sorted_matrix, expectation_signs[:, 0]))
+        elif j == 2:
+            sorted_matrix = np.column_stack((sorted_matrix, expectation_signs[:, 1]))
+        elif j == 3:
+            sorted_matrix = np.column_stack((sorted_matrix, expectation_signs[:, 1]))
+        elif j == 4:
+            sorted_matrix = np.column_stack((sorted_matrix, expectation_signs[:, 1]))
+        elif j == 5:
+            sorted_matrix = np.column_stack((sorted_matrix, expectation_signs[:, 2]))
+        elif j == 9:
+            sorted_matrix = np.column_stack((sorted_matrix, expectation_signs[:, 2]))
+        elif j == 13:
+            sorted_matrix = np.column_stack((sorted_matrix, expectation_signs[:, 2]))
+        else:
+            sorted_matrix = np.column_stack((sorted_matrix, expectation_signs[:, 3]))
+
+        temp = np.array(sorted_matrix[:, 2]) * np.array(sorted_matrix[:, 3])
+        sorted_matrix = np.column_stack((sorted_matrix, temp))
+
+        expectation_val = np.sum(temp)
+
+        return expectation_val
+
+    nqubits = 1
```

Comment:
do you really need the `nqubits` variable since it's always 1?

Reply:
I realised there were some redundant ```nqubits = ``` code in various functions. I've removed them. The only places with ```nqubits = 1``` or ```nqubits = 2``` are right at the top of the functions before the different sized matrices are specified.